Src/PerformanceCollector/Shared/Implementation/QuickPulse/Process/PerfLib/PerfLib.cs

```diff
@@ -0,0 +1,51 @@
+﻿namespace Microsoft.ApplicationInsights.Extensibility.PerfCounterCollector.Implementation.QuickPulse.PerfLib
+{
+    using System;
+    using System.Globalization;
+
+    internal class PerfLib : IQuickPulsePerfLib
+    {
+        private static PerfLib library = null;
+
+        private PerformanceMonitor performanceMonitor;
+
+        private PerfLib()
+        {
+        }
+
+        public static PerfLib GetPerfLib()
+        {
+            library = library ?? new PerfLib();
```

Comment:
Rather than checking and assigning to self each time you should just pre-create it.